docs/examples/gluster-s3-storage-template/README.md

```diff
@@ -2,111 +2,136 @@
 
 ## Prerequisites
 
-  OpenShift setup is up with master and nodes ready.
+* OpenShift setup is up with master and nodes ready.
 
-  cns-deploy tool is ran and heketi service is ready
+* cns-deploy tool has been run and heketi service is ready.
 
+## Deployment
 
+### 1. Provide the backend store
 
+The gluster-s3 service requires there be at least two GlusterFS volumes
+available for its use, one to store the object data and another for the
+object meta-data. In this example, we will create a new StorageClass to
+dynamically provision these two volumes on our pre-existing GlusterFS cluster.
 
-## First create a storageclass
+#### Create a StorageClass
 
-Create a GlusterFS StorageClass as below, replacing the `rest` parameters with your configuration:
+In our example, we have set up heketi to require a secret key for the admin user. A StorageClass created to use such a heketi instance needs a Secret that contains the admin key. This Secret is not needed if heketi is not configured to use an admin key.
+
+ Replace `NAMESPACE` and `ADMIN_KEY` parameters with your configuration.
+Here, `NAMESPACE` is the project and `ADMIN_KEY` is used for authorization to access Heketi service.
 
 ```
-oc create -f ./gluster-s3-storageclass.yaml
+oc create secret generic heketi-${NAMESPACE}-admin-secret --from-literal=key=${ADMIN_KEY} --type=kubernetes.io/glusterfs
 ```
 
-Available at 
-[gluster-s3-storageclass.yaml](./gluster-s3-storageclass.yaml)
+As an optional step, the Secret can be labelled. This is useful to be able to select the secret as part of a general query like `oc get
+--selector=glusterfs` and allows the secret to be removed programatically by the `gk-deploy` tool.
 
-## Start glusters3 service using template
+Replace `NAMESPACE` parameter with your configuration.
 
 ```
-oc new-app gluster-s3-template.yaml  --param=S3_ACCOUNT=testvolume  --param=S3_USER=adminuser --param=S3_PASSWORD=itsmine --param=VOLUME_CAPACITY=2Gi
+oc label --overwrite secret heketi-${NAMESPACE}-admin-secret glusterfs=s3-heketi-${NAMESPACE}-admin-secret gluster-s3=heketi-${NAMESPACE}-admin-secret
 ```
 
-Note: adjust parameters according to your needs.
+Create a GlusterFS StorageClass as below, replacing the `HEKETI_URL` and `NAMESPACE` parameters with your configuration.
+Also, provide `STORAGE_CLASS` to give a storage class name.
+`HEKETI_URL` is the REST url to access GlusterFs cluster.
+`NAMESPACE` is the project.
+`STORAGE_CLASS` is the StorageClass name provided by admin.
 
+```
+sed  -e 's/${HEKETI_URL}/heketi-store-project1.cloudapps.mystorage.com/g'  -e 's/${STORAGE_CLASS}/gluster-s3-store/g'      -e  's/${NAMESPACE}/store-project1/g'   deploy/ocp-templates/gluster-s3-storageclass.yaml | oc create -f -
```

Comment:
Remove the extra spaces before the `-e` flags.

Reply:
Done